tests/registration_test.py

```diff
@@ -533,6 +533,114 @@ def test_strings_groupby_attach(self):
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+
+        cleanup()
+        s = ak.array(["one", "two", "three", "123"])
+        s.register("sname")
+        ak.clear()
+        s.get_offsets()
+        sAttach = ak.Strings.attach("sname")
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+        s.get_bytes()
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+
+        cleanup()
+        s = ak.array(["one", "two", "three", "123"])
+        s.register("sname")
+        ak.clear()
+        s.get_bytes()
+        sAttach = ak.Strings.attach("sname")
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+        s.get_offsets()
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+
+        cleanup()
+        s = ak.array(["one", "two", "three", "123"])
+        # Testing for registration after being accessed and being removed after unregistering s
+        # Also tests for correct values in accessed arrays
+        s.register("sname")
+        ak.clear()
+        self.assertTrue(len(ak.list_registry()) == 1)
+
+        sbytes = s.get_bytes()
+        bytes_answers = ak.array(
+            [111, 110, 101, 0, 116, 119, 111, 0, 116, 104, 114, 101, 101, 0, 49, 50, 51, 0]
+        )
+        self.assertListEqual(sbytes.to_list(), bytes_answers.to_list())
+        self.assertTrue(len(ak.list_registry()) == 2)
+
+        soffsets = s.get_offsets()
+        offsets_answers = ak.array([0, 4, 8, 14])
+        self.assertListEqual(soffsets.to_list(), offsets_answers.to_list())
+        self.assertTrue(len(ak.list_registry()) == 3)
+
+        s.unregister()
+        self.assertTrue(len(ak.list_registry()) == 0)
+
+        # Tests for unregistering each property
+        s = ak.array(["one", "two", "three", "123"])
+        s.register("sname")
+        sbytes = s.get_bytes()
+        soffsets = s.get_offsets()
+        self.assertTrue(len(ak.list_registry()) == 3)
+        sbytes.unregister()
+        self.assertTrue(len(ak.list_registry()) == 2)
+        soffsets.unregister()
+        self.assertTrue(len(ak.list_registry()) == 1)
```

Comment:
I would also check that in addition to the length of the registry, we check that the registered name is not in `list_registry`